### ISSUE TITLE
Wrong django's TextChoices.value after 1.1.307

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#5092]
First of all, the expansion of `Enum.value` to its literal options is really nice!

But I just faced an issue with a django's TextChoices: https://docs.djangoproject.com/en/4.2/ref/models/fields/#enumeration-types

Those are implemented as a subclass of enum, but supports receiving the value as a tuple of `(value, label)`.

With 1.1.307 this code produces the following:

```python
from django.db import models


class MyEnum(models.TextChoices):
    FOO = "foo", "Foo option"
    BAR = "bar", "bar option"

# obs. the correct type for this would be Literal['foo'] (or at least a generic str)
reveal_type(MyEnum.FOO.value)  # Type of "MyEnum.FOO.value" is "tuple[Literal['foo'], Literal['Foo option']]"
# this is still works fine
reveal_type(MyEnum.FOO.label)  # Type of "MyEnum.FOO.label" is "str"

def foo(x: MyEnum):
    # obs. the correct type for this would be Literal['foo'] | Literal['bar'] (or at least a generic str)
    reveal_type(x.value)  # Type of "x.value" is "tuple[Literal['foo'], Literal['Foo option']] | tuple[Literal['bar'], Literal['bar option']]"
    # this is still works fine
    reveal_type(x.label)  # Type of "x.label" is "str"
```

Those seems to be even overriding the typing defined in the stubs that I'm using in here: https://github.com/sbdchd/django-types/blob/main/django-stubs/db/models/enums.pyi#L36

Don't know if this is something that pyright can workaround, but if it is not, how can the stub typing be fixed in order to support this?

**Comment by erictraut** [comment]
Can you please provide a self-contained code sample? Your code above is referencing undefined symbols like `model`.

**Comment by erictraut** [comment]
Ah, looks like it just requires the addition of this import statement: `from django.db import models`.

**Comment by bellini666** [comment]
> Ah, looks like it just requires the addition of this import statement: `from django.db import models`.

Yes, sorry! And also install the `django-types` package in the venv you are testing

**Comment by erictraut** [comment]
I've modified the type inference logic for the `value` attribute. If it detects a custom enum metaclass, it infers the `value` type as `Any`.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.308, which I just published. It will also be included in a future release of pylance.
